### PR TITLE
docs(dns) add extra documentation on the DNS

### PR DESCRIPTION
**Pull request by lahabana**
CoreDNS wasn't mentionned anywhere, explain how DPP DNS works
and how to update the template

Signed-off-by: Charly Molter <charly.molter@konghq.com>

### CODE REVIEW

**Review by Bradamant3 — CHANGES_REQUESTED** [review]
In addition to suggestions and inline questions, I wonder whether it might be more useful to put the `How it works` section at the top of the Data plane proxy DNS section, and then leave the `Override ...` section where it is. wdyt?

File: docs/docs/1.3.0/networking/dns.md [hunk 1]
```diff
@@ -1,6 +1,6 @@
 # DNS
 
-In Kuma version , DNS on the data plane proxy is enabled by default on Kubernetes. You can also continue to deploy with [DNS on the control plane](#control-plane-dns).
+In Kuma version 1.3.0, DNS on the data plane proxy is enabled by default on Kubernetes. You can also continue to deploy with [DNS on the control plane](#control-plane-dns).
```
[the remark this answers is not in the document]
Comment (lahabana) [reply docs/docs/1.3.0/networking/dns.md:3]: Shouldn't the wording be "Since" then?

[the remark this answers is not in the document]
Comment (lahabana) [reply docs/docs/1.3.0/networking/dns.md:3]: Saw you've changed this to `As of` which is the same as Since!

**Comment by lahabana** [comment]
I'm unsure about your suggestion I feel like you don't need to understand it to set it up.
You need it for advance use cases which is why it's lower down in the docs.

Otherwise I did all the updates you suggested and sorry about the future tense (bad habits die hard!)

**Review by Bradamant3 — APPROVED** [review]

**Comment by Bradamant3** [comment]
Heh future tense is all over the place and I'm not consistent about correcting it either. Iterate, iterate ...

Point of clarification (I might revisit in a separate PR) -- I made the suggestion to move the explanation of how DNS works up the file to avoid duplication, not to explain something users don't really need :D. So I'll go back to the file as a whole and rethink organization generally (this is a common issue throughout the docs, not limited to this page).

I'm also not seeing commits for some of the suggestions I made, but they aren't a big deal -- we can revisit another time ...

**Comment by lahabana** [comment]
I took the suggestions in the general commit where other suggestions were added.

**Pull request merged**